tests/unit/test_serial_number.cpp

```diff
@@ -65,6 +66,48 @@ template <typename UintType> void TestSerialNumber(const char *aName)
     printf("TestSerialNumber<%s>() passed\n", aName);
 }
 
+void TestMinMaxClamp(void)
+{
+    uint16_t u16;
+    uint32_t u32;
+
+    VerifyOrQuit(Min<uint8_t>(1, 2) == 1);
+    VerifyOrQuit(Min<uint8_t>(2, 1) == 1);
+    VerifyOrQuit(Min<uint8_t>(1, 1) == 1);
+
+    VerifyOrQuit(Max<uint8_t>(1, 2) == 2);
+    VerifyOrQuit(Max<uint8_t>(2, 1) == 2);
+    VerifyOrQuit(Max<uint8_t>(1, 1) == 1);
+
+    VerifyOrQuit(Clamp<uint8_t>(1, 5, 10) == 5);
+    VerifyOrQuit(Clamp<uint8_t>(5, 5, 10) == 5);
+    VerifyOrQuit(Clamp<uint8_t>(7, 5, 10) == 7);
+    VerifyOrQuit(Clamp<uint8_t>(10, 5, 10) == 10);
+    VerifyOrQuit(Clamp<uint8_t>(12, 5, 10) == 10);
```

Comment:
Add test for when min value is greater than max value?

Reply:
For `Clamp()` if caller uses `aMin > aMax` we consider it undefined behavior (same as [`std::clamp()`](https://en.cppreference.com/w/cpp/algorithm/clamp))  (since we do `Min(Max(aValue, aMin), aMax)` the result will be always `aMax`). 

One interesting case is if `aMin` and `aMax` are equal. Will add this in the test.